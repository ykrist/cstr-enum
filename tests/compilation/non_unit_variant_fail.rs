use cstr_enum::*;

#[derive(FromCStr)]
enum Enum0 {
  A(u8),
  B,
  C
}

#[derive(FromCStr)]
enum Enum1 {
  A,
  B{ foo: u8, bar: u16},
  C
}

fn main() {

}