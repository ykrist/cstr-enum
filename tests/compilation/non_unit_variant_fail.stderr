error: variant cannot have fields
 --> $DIR/non_unit_variant_fail.rs:5:3
  |
5 |   A(u8),
  |   ^^^^^

error: variant cannot have fields
  --> $DIR/non_unit_variant_fail.rs:13:3
   |
13 |   B{ foo: u8, bar: u16},
   |   ^^^^^^^^^^^^^^^^^^^^^
